internal/provider/vaultsecrets/resource_vault_secrets_secret.go

```diff
@@ -76,6 +76,9 @@ func (r *resourceVaultsecretsSecret) Schema(_ context.Context, _ resource.Schema
 						"must contain only ASCII letters, numbers, and underscores; must not start with a number",
 					),
 				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
```

Comment:
We should also add this to the app_name since it is not a modifiable field.

Can we add tests for that scenario? We can either use the TF test framework to verify changing the secret_name causes a recreate operation or add a TestConfigFunc that verifies that the secret with the old name no longer exists after the rename is applied.